### PR TITLE
View refactor

### PR DESCRIPTION
**Pull request by dwang733**
We refactored most of the view files for styling. We also came up with some basic style guides for the views:

- Avoid using %div. If there's a class or id associated, using that will automatically insert a div.
- When creating hashes, please use "key: value" notation instead of ":key => value" notation.
- Prefer single quotes over double quotes, except when necessary (e.g. string interpolation like "#{value}")
- When render partials, please use the notation "render 'partial_name', with local variables being passed as key: value. Avoid using the old notation of "render partial: 'partial_name', locals: {key: value}".
- Make sure that CSS IDs are not duplicated on a page (especially in loops). Use CSS classes instead.

This PR will not be merged until the constrainCarouselElements branch is merged.

### CODE REVIEW

**Comment by Juggernaut265** [comment]
This seems fine, but, because of major changes to "discovery_controller.rb", "models/vendor.rb", and "views/discovery/_index.html.haml", it might be good to temporarily remove them from this PR because they might clash with the changes. I can update the style if needed in my own PR.

File: app/views/application/_header.html.haml [hunk 1]
```diff
@@ -1,32 +1,32 @@
...
-  %div{:class => "navbar navbar-expand-lg navbar-dark bg-dark static-top"}
-    %div{:class => "container-fluid"}
-      %div{:class => "navbar-header"}
-        = link_to discovery_path, id: "logo", class: "navbar-brand w-50" do
-          = image_tag "logo450x429.png", id: "logoheader"
-      %button{:type => "button", :class => "navbar-toggler", :data => {:toggle => "collapse", :target => "#navbarResponsive"}, :aria => {:expanded => "false", :controls => "navbarResponsive", :label => "Toggle navigation"}}
-        %span{:class => "navbar-toggler-icon"}
-      %div{:id => "navbarResponsive", :class => "collapse navbar-collapse w-100"}
-        %ul{:class => "navbar-nav w-100"}
+  .navbar.navbar-expand-lg.navbar-dark.bg-dark.static-top
```
Comment (nrleuenb) [inline app/views/application/_header.html.haml:2]: Is there a way to limit this in any way? It's a lot of classes for one line

File: app/views/discovery/_carousel_vendor_element.html.haml [hunk 2]
```diff
@@ -1,7 +0,0 @@
-%div{:class => "container carousel_element #{local_assigns[:ownership].gsub(" ", "_")}", :id => local_assigns[:vendor].name.gsub(" ", "_"), :onclick => "routeToPageDirectly('#{vendor_path(local_assigns[:vendor].id)}')"}
```
Comment (nrleuenb) [inline app/views/discovery/_carousel_vendor_element.html.haml:1]: This has been replaced with the generic carousel_element correct?

**Review by nrleuenb — APPROVED** [review]
LGTM. 

This branch breaks discovery-ui, fyi, so it needs to be merged ASAP

File: app/views/application/_header.html.haml (hunk 1 above)
Comment (jlabowitz) [reply app/views/application/_header.html.haml:2]: These classes have always been here. We just rearranged them differently. I am not sure if there are ways to shorten this line, but that is not the purpose of this PR.

File: app/views/discovery/_carousel_vendor_element.html.haml (hunk 2 above)
Comment (jlabowitz) [reply app/views/discovery/_carousel_vendor_element.html.haml:1]: Yes, that's correct. David found that _carousel_vendor_element and _carousel_product_element were essentially the same thing and easily factored them into the same partial.

**Comment by jlabowitz** [comment]
Looks good after handling the merge conflicts.

**Pull request merged**